sdk/src/types/block/slot/index.rs

```diff
@@ -3,16 +3,24 @@
 
 use derive_more::{Deref, Display, From};
 
+use crate::utils::serde::string;
+
 /// Timeline is divided into slots, and each slot has a corresponding slot index.
 /// To calculate the slot index of a timestamp, `genesisTimestamp` and the duration of a slot are needed.
 /// The slot index of timestamp `ts` is `(ts - genesisTimestamp)/duration + 1`.
-#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash, From, Deref, Display, packable::Packable)]
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash, From, Deref, Display, PartialOrd, Ord, packable::Packable)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
```

Comment:
I think we need a `serde(transparent)` here.

Reply:
I've added, but I'm wondered if its necessary for 'one-level' struct new types.

Reply:
It may not be for the tuple. Can you test to be sure?

Reply:
ok, I know why I had this impression. Most of serializers/deserializers ignore the new-type and use serialization for the inner type. https://github.com/serde-rs/json/blob/ab084832926cf7d733785c8801aa4fcf96523956/src/ser.rs#L221  . So it's better to use `transparent` to avoid relying on the implementation of the serializer.

Reply:
Agreed